core/validation/specific_validations/towns_fund_round_four.py

```diff
@@ -349,57 +349,74 @@ def validate_funding_spent(workbook: dict[str, pd.DataFrame]) -> list["TownsFund
-                    + (f" - Project {get_project_number_by_id(idx, project_ids)}" if validate_by == "project" else ""),
+                    section="Project Funding Profiles",
                     column="Grand Total",
-                    message=msgs.OVERSPEND_PROJECT if validate_by == "project" else msgs.OVERSPEND_PROGRAMME,
+                    message=msgs.OVERSPEND_PROGRAMME,
                     # first project begins after 17 and each project is seperated by 28 cells
-                    row_indexes=[17 + 28 * get_project_number_by_id(idx, project_ids)]
-                    if validate_by == "project"
-                    else [17 + 28 * get_project_number_by_id(proj_id, project_ids) for proj_id in project_ids],
+                    row_indexes=[17 + 28 * get_project_number_by_id(proj_id, project_ids) for proj_id in project_ids],
                 )
             )
 
+    else:
+        # check funding against individual project funding allocated for Towns Deal submissions
+        ids_to_check = project_ids
+        for expense_type in ["CDEL", "RDEL"]:
+            for idx in ids_to_check:
+                project_number = get_project_number_by_id(idx, project_ids)
+                if funding_spent[expense_type][idx] > DefaultConfig.TF_FUNDING_ALLOCATED[expense_type][idx]:
+                    funding_spent_failures.append(
+                        TownsFundRoundFourValidationFailure(
+                            sheet="Funding",
+                            section=f"Project Funding Profiles - Project {project_number}",
+                            column="Grand Total",
+                            message=msgs.OVERSPEND_CDEL if expense_type == "CDEL" else msgs.OVERSPEND_RDEL,
+                            row_indexes=[13 + 28 * project_number]
+                            if expense_type == "CDEL"
+                            else [16 + 28 * project_number],
+                        )
+                    )
+
     return funding_spent_failures
 
 
-def spend_per_project(funding_df: pd.DataFrame, project_id: str) -> float:
+def spend_per_project(funding_df: pd.DataFrame, project_id: str) -> dict[str:int]:
     """return the total funding spent per an individual project
 
     :param funding_df: A dataframe of the funding table from a submission
     :param: project_id: ID of project in question
 
     :return: Total funding spent per project
     """
-    funding_spent = (
-        funding_df["Spend for Reporting Period"]
-        .loc[
-            (funding_df["Project ID"] == project_id)
-            & (funding_df["Funding Source Type"] == "Towns Fund")
-            & ~(funding_df["Funding Source Name"].str.contains(r"contractually committed"))
-        ]
-        .sum()
-    )
+    funding_df = funding_df.loc[
+        (funding_df["Project ID"] == project_id)
+        & (funding_df["Funding Source Type"] == "Towns Fund")
+        & ~(funding_df["Funding Source Name"].str.contains(r"contractually committed"))
```

Comment:
Out of curiosity, why are these raw strings?

Reply:
Not sure, but they didn't need to be so changed